src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/AbstractConverter.cs

```diff
@@ -93,5 +114,158 @@ private static QueryClauseSyntax CreateQueryClause(ExtendedSyntaxNode node)
+            int currentExtendedNodeIndex = 0;
+
+            return CreateLinqInvocation(
+                foreachStatement,
+                receiverForInvocation: foreachStatement.Expression,
+                selectExpression: selectExpression,
+                leadingCommentsTrivia: ForEachInfo.LeadingTokens.GetTrivia(),
+                trailingCommentsTrivia: Enumerable.Empty<SyntaxTrivia>(),
+                currentExtendedNodeIndex: ref currentExtendedNodeIndex)
+                .WithAdditionalAnnotations(Formatter.Annotation);
+        }
+
+        private InvocationExpressionSyntax CreateLinqInvocation(
+            ForEachStatementSyntax forEachStatement,
+            ExpressionSyntax receiverForInvocation,
+            IEnumerable<SyntaxTrivia> leadingCommentsTrivia,
+            IEnumerable<SyntaxTrivia> trailingCommentsTrivia,
+            ExpressionSyntax selectExpression,
+            ref int currentExtendedNodeIndex)
+        {
+            leadingCommentsTrivia = forEachStatement.ForEachKeyword.GetAllTrivia().Concat(leadingCommentsTrivia);
+
+            // Recursively create linq invocations, possibly updating the receiver (Where invocations), to get the inner expression for
+            // the lambda body for the linq invocation to be created for this foreach statement. For example:
+            //
+            // INPUT:
+            //   foreach (var n1 in c1)
+            //      foreach (var n2 in c2)
+            //          if (n1 > n2)
+            //              yield return n1 + n2;
+            //
+            // OUTPUT:
+            //   c1.SelectMany(n1 => c2.Where(n2 => n1 > n2).Select(n2 => n1 + n2))
+            //
+            bool hasForEachChild = false;
+            var lambdaBody = CreateLinqInvocationForExtendedNode(selectExpression, ref currentExtendedNodeIndex, ref receiverForInvocation, ref hasForEachChild);
+            var lambda = SyntaxFactory.SimpleLambdaExpression(
+                SyntaxFactory.Parameter(
+                    forEachStatement.Identifier.WithPrependedLeadingTrivia(
+                    SyntaxNodeOrTokenExtensions.GetTrivia(forEachStatement.Type.GetFirstToken())
+                        .FilterComments(addElasticMarker: false))),
+                lambdaBody)
+                .WithCommentsFrom(leadingCommentsTrivia, trailingCommentsTrivia,
+                    forEachStatement.OpenParenToken, forEachStatement.InKeyword, forEachStatement.CloseParenToken);
+
+            // Create Select or SelectMany linq invocation for this foreach statement. For example:
+            //
+            // INPUT:
+            //   foreach (var n1 in c1)
+            //      ...
+            //
+            // OUTPUT:
+            //   c1.Select(n1 => ...
+            //      OR
+            //   c1.SelectMany(n1 => ...
+            //
+            var invokedMethodName = !hasForEachChild ? "Select" : "SelectMany";
```

Comment:
nameof

Reply:
attempting to use `nameof(IEnumerable<int>.Select)` generates an error:
```
CS8093	Extension method groups are not allowed as an argument to 'nameof'.
```

Reply:
`nameof(Enumerable.Select)`